### ISSUE TITLE
Propagate `Double64` through `sign` function

### ISSUE THREAD

**Issue by DanielDoehring** [JuliaMath/DoubleFloats.jl#209]
```julia
julia> typeof(sign(Double64(42)))
Float64
```
In my opinion, the `sign` function should return an integer, but as the standard/Base Julia version seems to return a Float, I would like to see this propagating through.

**Comment by JeffreySarnoff** [comment]
you want `sign(Double64(_))` to return a Double64?

**Comment by DanielDoehring** [comment]
Yes, to match the expected [behaviour as documented](https://docs.julialang.org/en/v1/base/math/#Base.sign) for floating point datatypes :+1:
